Add development notes to README.md

Blarc opened the pull request:
I've had to figure out the dependencies needed for local development so I thought I would add them to the `README.md` to make it easier for contributing. I listed the dependencies I needed to install before I could run kubedock via `make run` command.

joyrex2001 commented:
Are you developing on linux (since you require the btrfs dependencies)? In the build I disabled cgo for this, maybe this also helps during development? Probably adding `CGO_ENABLED=0` before the go run in the `run` task in the `Makefile` will make sure you don't need the dependencies.

Blarc commented:
You are correct. I uninstalled the dependencies and it ran with `CGO_ENABLED=0`. I have updated the `README.md` and added the set `CGO_ENABLED=0` in the Makefile. I guess the content in the README.md might be a bit redundant now :thinking: .

joyrex2001 commented:
Don't think the `go get .` is required either, I don't think the readme section adds that much value at this point (as it's basically just do a `make run`. I do want to keep the change to the Makefile though; feel free to remove the readme part, and I'll merge.

Blarc commented:
I dropped the commit that added the readme part.

joyrex2001 commented:
Thanks! 👍🏻
